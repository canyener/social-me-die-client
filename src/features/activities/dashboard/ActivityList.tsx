import React from 'react'
import { Item, Button, Label, Segment } from 'semantic-ui-react'

import { IActivity } from '../../../app/models/activity'

interface IProps {
  activities: IActivity[]
}

export const ActivityList: React.FC<IProps> = ({ activities }) => (
  <Segment clearing>
    <Item.Group divided>
      {
        activities.map(activity =>
          <Item key={activity.id}>
            <Item.Content>
              <Item.Header as='a'>{activity.title}</Item.Header>
              <Item.Meta>{activity.date.toLocaleString('tr-TR', { timeZone: 'UTC' })}</Item.Meta>
              <Item.Description>
                <div>{activity.description}</div>
                <div>{activity.city}, {activity.venue}</div>
              </Item.Description>
              <Item.Extra>
                <Button floated='right' content='View' color='blue' />
                <Label basic content={activity.category} />
              </Item.Extra>
            </Item.Content>
          </Item>
        )
      }
    </Item.Group>
  </Segment>
)
