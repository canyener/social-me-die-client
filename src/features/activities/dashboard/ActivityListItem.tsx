import React from 'react'
import { Item, Button, Segment, Icon } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

import { IActivity } from '../../../app/models/activity'

const ActivityListItem: React.FC<{ activity: IActivity }> = ({ activity }) => {
  return (
    <Segment.Group>
      <Segment>
        <Item.Group>
          <Item>
            <Item.Image size='tiny' circular src='/assets/user.png' />
            <Item.Content>
              <Item.Header as='a'>{activity.title}</Item.Header>
              <Item.Description>
                Hosted by Cancan
            </Item.Description>
            </Item.Content>
          </Item>
        </Item.Group>
      </Segment>
      <Segment>
        <Icon name='clock' /> {activity.date!.toLocaleDateString('tr-TR')}
        <Icon name='marker' /> {activity.venue}, {activity.city}
      </Segment>
      <Segment secondary>
        Attendees will go here
      </Segment>
      <Segment clearing>
        <span>{activity.description}</span>
        <Button
          as={Link}
          to={`/activities/${activity.id}`}
          floated='right'
          content='View'
          color='blue'
        />
      </Segment>
    </Segment.Group>
  )
}

export default ActivityListItem