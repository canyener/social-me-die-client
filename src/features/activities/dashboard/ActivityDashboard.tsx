import React from 'react'
import { Grid } from 'semantic-ui-react'
import { observer } from 'mobx-react-lite'

import ActivityList from './ActivityList'

const ActivityDashboard: React.FC = () => {
  return (
    <Grid>
      <Grid.Column width={10}>
        <ActivityList />
      </Grid.Column>
      <Grid.Column width={6}>
        <h2>Activity Filters</h2>
      </Grid.Column>
    </Grid>
  )
}

ActivityDashboard.displayName = 'ActivityDashboard'

export default observer(ActivityDashboard)