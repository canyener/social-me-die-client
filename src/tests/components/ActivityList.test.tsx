import React from 'react'
import { shallow } from 'enzyme'

import { ActivityList } from '../../features/activities/dashboard/ActivityList'
import { fakeActivities } from '../fixtures/seed'

test('Should render ActivityList correctly', () => {
  const wrapper = shallow(
    <ActivityList
      activities={fakeActivities}
      selectActivity={() => { }}
      deleteActivity={() => { }}
    />)

  expect(wrapper).toMatchSnapshot()
})

test('Should call selectActivity once when button is clicked', () => {
  const selectActivitySpy = jest.fn()

  const wrapper = shallow(
    <ActivityList
      activities={fakeActivities}
      selectActivity={selectActivitySpy}
      deleteActivity={() => { }}
    />)
  wrapper.find('Button').first().simulate('click')
  expect(selectActivitySpy).toHaveBeenCalledTimes(1)
})

test('Should call deleteActivity once when delete button is clicked', () => {
  const deleteActivitySpy = jest.fn()
  const wrapper = shallow(
    <ActivityList
      activities={fakeActivities}
      selectActivity={() => { }}
      deleteActivity={deleteActivitySpy}
    />)

    wrapper.find('Button').at(1).simulate('click')
    expect(deleteActivitySpy).toHaveBeenCalledTimes(1)
})