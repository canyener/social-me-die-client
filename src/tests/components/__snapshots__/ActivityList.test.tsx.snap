// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ActivityList correctly 1`] = `
<Segment
  clearing={true}
>
  <ItemGroup
    divided={true}
  >
    <Item
      key="1"
    >
      <ItemContent>
        <ItemHeader
          as="a"
        >
          Fake Activity 1
        </ItemHeader>
        <ItemMeta>
          12/31/1969, 7:00:01 PM
        </ItemMeta>
        <ItemDescription>
          <div>
            Fake description
          </div>
          <div>
            Fake City
            , 
            Fake Venue
          </div>
        </ItemDescription>
        <ItemExtra>
          <Button
            as="button"
            color="blue"
            content="View"
            floated="right"
            onClick={[Function]}
          />
          <Label
            basic={true}
            content="Fake Category 1"
          />
        </ItemExtra>
      </ItemContent>
    </Item>
    <Item
      key="2"
    >
      <ItemContent>
        <ItemHeader
          as="a"
        >
          Fake Activity 2
        </ItemHeader>
        <ItemMeta>
          12/31/1969, 7:00:01 PM
        </ItemMeta>
        <ItemDescription>
          <div>
            Fake description 2
          </div>
          <div>
            Fake City 2
            , 
            Fake Venue 2
          </div>
        </ItemDescription>
        <ItemExtra>
          <Button
            as="button"
            color="blue"
            content="View"
            floated="right"
            onClick={[Function]}
          />
          <Label
            basic={true}
            content="Fake Category 2"
          />
        </ItemExtra>
      </ItemContent>
    </Item>
  </ItemGroup>
</Segment>
`;
